### PR TITLE
make debounce and throttle time configurable

### PR DESCRIPTION
**Pull request by MurdeRM3L0DY**
supersedes #639.

### CODE REVIEW

File: doc/cmp.txt [hunk 1]
```diff
@@ -366,6 +366,16 @@ enabled~
   `boolean | fun(): boolean`
   Toggles the plugin on and off.
 
+                                                *cmp-config.performance.debounce*
+performance.debounce~
+  `number`
+  Sets debounce time
```
Comment (hrsh7th) [inline doc/cmp.txt:372]: I think we should add more obvious description for this.

File: doc/cmp.txt [hunk 2]
```diff
@@ -366,6 +366,16 @@ enabled~
...
+
+                                                *cmp-config.performance.throttle*
+performance.throttle~
+  `number`
+  Sets throttle time
```
Comment (hrsh7th) [inline doc/cmp.txt:377]: ditto

This value is used for throttling `filtering` and `displaying` results.
For example, nvim-cmp shouldn't filter results 7 times if the user press `console` very fast.
If the user specify `throttle_time` as 120(ms), nvim-cmp will filter results each 70ms.

File: doc/cmp.txt (hunk 1 above)
Comment (hrsh7th) [reply doc/cmp.txt:372]: Hm... the `debounce` is hard to explain.

If the user press `console` and it's invoking 10 number of sources, the nvim-cmp will retrieve completion results 10 times.

Typically, the each responses would come back from sources in small milliseconds.
I think it should be merged as one filtering process. The debounce_times is used for this.

Comment (MurdeRM3L0DY) [reply doc/cmp.txt:372]: So debounce is basically how long cmp waits to retrieve completion results from sources before displaying.

**Comment by MurdeRM3L0DY** [comment]
WDYT? @hrsh7th

**Comment by hrsh7th** [comment]
I can merge this after docmentation updated.

**Comment by hrsh7th** [comment]
LGTM. Thank you~!

**Comment by MurdeRM3L0DY** [comment]
> LGTM. Thank you~!

Happy to contribute. Thank YOU for this amazing plugin!

**Comment by hrsh7th** [comment]
Yes. we can close that. I'll do.

**Pull request merged**